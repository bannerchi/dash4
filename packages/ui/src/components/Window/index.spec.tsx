/// <reference types="@types/jest" />
import React from 'react';
import { render } from 'react-testing-library';
import { Window, WindowBody, WindowHeader } from './index';

describe('Window', () => {
	test('exist', () => {
		expect(typeof Window).toBe('object');
	});
	test('render default', () => {
		const { container } = render(<Window>Hello</Window>);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render with className', () => {
		const { container } = render(<Window className="test">Hello</Window>);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render dark', () => {
		const { container } = render(<Window dark>Hello</Window>);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render fullscreen', () => {
		const { container } = render(<Window fullscreen>Hello</Window>);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render !autoStretch', () => {
		const { container } = render(<Window autoStretch={false}>Hello</Window>);
		expect(container.firstChild).toMatchSnapshot();
	});
});
