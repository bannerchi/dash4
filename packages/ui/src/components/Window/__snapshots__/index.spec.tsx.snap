// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Window render !autoStretch 1`] = `
<div
  class="Component-window-0-1-1 Component-windowBright-0-1-3"
>
  Hello
</div>
`;

exports[`Window render dark 1`] = `
<div
  class="Component-window-0-1-1 Component-windowDark-0-1-4 Component-windowAutoStretch-0-1-2"
>
  Hello
</div>
`;

exports[`Window render default 1`] = `
<div
  class="Component-window-0-1-1 Component-windowBright-0-1-3 Component-windowAutoStretch-0-1-2"
>
  Hello
</div>
`;

exports[`Window render fullscreen 1`] = `
<div
  class="Component-window-0-1-1 Component-windowBright-0-1-3 Component-windowAutoStretch-0-1-2 Component-windowFullscreen-0-1-5"
>
  Hello
</div>
`;

exports[`Window render with className 1`] = `
<div
  class="Component-window-0-1-1 Component-windowBright-0-1-3 test Component-windowAutoStretch-0-1-2"
>
  Hello
</div>
`;
